src/EnergyPlus/BranchNodeConnections.cc

```diff
@@ -665,6 +665,7 @@ namespace BranchNodeConnections {
 					}
 				}
 				if ( ! IsValid ) {
+					if (NodeConnections(Loop1).ObjectType == "COILSYSTEM:INTEGRATEDHEATPUMP:AIRSOURCE") continue;
```

Comment:
if we didn’t bypass this, it would cause fatal error and never allow a run-through. You see the ASIHP violates some “fundamental” E+ rules, i.e. same air and water nodes are connected to many different coil objects, and more than one parent objects refer to the same ASIHP object, since ASIHP is a multi-functional object for space conditioning and water heating. Probably, bypassing these checks is an easier way to implement the ASIHP function without breaking too many E+ present logics and causing a major modification.